<launch>
    <!-- 参数设置 -->
    <arg name="drone_id"/>
    <arg name="origin_lat"/>
    <arg name="origin_lon"/>
    <arg name="origin_alt"/>
    <arg name="init_x"/>
    <arg name="init_y"/>
    <arg name="init_z"/>
    
    <!-- ESKF 节点 -->
    <node pkg="co_navi_pkg" type="IMU_GPS_eskf_fusion" name="IMU_GPS_fusion_$(arg drone_id)" output="screen">
        <param name="drone_id" value="$(arg drone_id)"/>
        <!-- 噪声参数 -->     
        <param name="sigma_g" value="0.005" />
        <param name="sigma_a" value="0.05" />
        <param name="sigma_p" value="0.3" />
        <param name="sigma_vgps" value="0.25" />
        <param name="sigma_m" value="5e-6" />

        <param name="origin_lat" value="$(arg origin_lat)" />
        <param name="origin_lon" value="$(arg origin_lon)" />
        <param name="origin_alt" value="$(arg origin_alt)" />
        <param name="init_x" value="$(arg init_x)" />
        <param name="init_y" value="$(arg init_y)" />
        <param name="init_z" value="$(arg init_z)" />
    </node>

</launch>
